sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -562,20 +575,26 @@ class FilesElement extends Polymer.Element {
         .then((closeResult: InputDialogCloseResult) => {
           if (closeResult.confirmed && closeResult.userInput) {
             const newName = this.currentPath + '/' + closeResult.userInput;
+
+            let success: boolean;
             return ApiManager.renameItem(selectedObject.path, newName)
-              .then(() => this._fetchFileList())
               .then(() => {
+                success = true;
                 // Dispatch a success notification
                 const message = 'Renamed ' + selectedObject.name +
                     ' to ' + closeResult.userInput + '.';
                 this.dispatchEvent(new NotificationEvent(message));
-              });
-              // TODO: [yebrahim] Re-select the renamed item after refresh
+              })
+              .catch((e: Error) => {
+                success = false;
+                Utils.showErrorDialog('Error renaming item', e.message);
+              })
+                // Only refresh the file list on success
+              .then(() => success === true ? this._fetchFileList() : null);
```

Comment:
Yes, I like that better. Since you are doing the fetchFileList-and-catch something like five times, how about adding a method fetchFileListAndIgnoreErrors that encapsulates that additional catch?